File: src/guide/models/session.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from .shared import EventSignificance, EventType, Perspective


class Session(BaseModel):
    id: UUID
    campaign_id: UUID
    session_number: int = 1
    title: str | None = None
    notes: str | None = None
    started_at: datetime | None = None
    ended_at: datetime | None = None
    created_at: datetime
    updated_at: datetime


class SessionEvent(BaseModel):
    id: UUID
    session_id: UUID
    campaign_id: UUID
    event_type: EventType
    description: str
    significance: EventSignificance = EventSignificance.minor
    is_player_visible: bool = True
    involved_character_ids: list[UUID] = Field(default_factory=list)
    occurred_at: datetime


class SessionSummary(BaseModel):
    session_id: UUID
    perspective: Perspective
    content: str
    generated_at: datetime


class CreateSessionRequest(BaseModel):
    title: str | None = None
    notes: str | None = None


class CreateSessionEventRequest(BaseModel):
    event_type: EventType
    description: str
    significance: EventSignificance | None = None
    is_player_visible: bool | None = None
    involved_character_ids: list[UUID] | None = None

File: crates/guide-core/src/models/session.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

use super::shared::{EventSignificance, EventType};

#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
pub struct Session {
    pub id: Uuid,
    pub campaign_id: Uuid,
    pub session_number: i32,
    pub title: Option<String>,
    pub notes: Option<String>,
    pub started_at: Option<DateTime<Utc>>,
    pub ended_at: Option<DateTime<Utc>>,
    pub map_url: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
pub struct SessionEvent {
    pub id: Uuid,
    pub session_id: Uuid,
    pub campaign_id: Uuid,
    pub event_type: EventType,
    pub description: String,
    pub significance: EventSignificance,
    pub involved_character_ids: Vec<Uuid>,
    pub occurred_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
pub struct SessionSummary {
    pub session_id: Uuid,
    pub content: String,
    pub generated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
pub struct CreateSessionRequest {
    pub title: Option<String>,
    pub notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
pub struct CreateSessionEventRequest {
    pub event_type: EventType,
    pub description: String,
    pub significance: Option<EventSignificance>,
    pub involved_character_ids: Option<Vec<Uuid>>,
}

#[derive(Debug, Clone, Serialize, Deserialize, utoipa::ToSchema)]
pub struct ImprovPromptResponse {
    pub options: Vec<String>,
}
